sdk/src/secp256k1.rs

```diff
@@ -72,7 +72,7 @@ pub fn verify_eth_addresses(
 
         // Parse out signature
         let signature_index = offsets.signature_instruction_index as usize;
-        if signature_index > instruction_datas.len() {
+        if signature_index >= instruction_datas.len() {
```

Comment:
I also think adding a test for this won't hurt us, considering this is **the signature verification code**. Maybe this should be easy?

Reply:
This actually a more complete fix and adds tests:
https://github.com/solana-labs/solana/pull/13026